### PR TITLE
Staging and production environments

### PR DESCRIPTION
**Pull request by dr-bizz**
Adding Staging and production environments, to make it easier to switch between the two locally.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -39,7 +39,9 @@ jobs:
         run: yarn lint
 
       - name: 🏗️ Build app
-        run: yarn build --configuration production
+        run: yarn build --configuration $buildEnvironment
+        env: 
+          buildEnvironment: ${{ github.ref == 'refs/heads/staging' && 'staging' || 'production' }}
```
Comment (frett) [inline .github/workflows/node.js.yml:44]: for the checks & build test I would always have it build production, that's the important system to make sure things build correctly for

File: src/app/new-page/page.component.ts [hunk 2]
```diff
@@ -248,8 +249,9 @@ export class PageNewComponent implements OnInit, OnDestroy {
     ) {
       const manifestName = item.attributes['manifest-name'];
       const translationid = item.id;
-      const fileName =
-        APIURL.GET_XML_FILES_FOR_MANIFEST + translationid + '/' + manifestName;
+      const fileName = environment.production
+        ? APIURL.GET_XML_FILES_FOR_MANIFEST + translationid + '/' + manifestName
```
Comment (frett) [inline src/app/new-page/page.component.ts:253]: I can't wait to be able to get rid of this for production

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -68,7 +69,9 @@ jobs:
         run: yarn install
 
       - name: 🏗️ Build app
-        run: yarn build --configuration production
+        run: yarn build --configuration $buildEnvironment
+        env: 
+          buildEnvironment: ${{ github.ref == 'refs/heads/development' && 'staging' || 'production' }}
```
Comment (frett) [inline .github/workflows/node.js.yml:74]: should `staging` here be `development`?

Comment (dr-bizz) [reply .github/workflows/node.js.yml]: Good catch, I altered it on the development branch and not on this PR. Should be fixed now.

**Pull request merged**